docs/general/dql/selects.rst

```diff
@@ -511,274 +502,461 @@ does always return ``NULL`` when comparing ``NULL``.
+    +----------+-------------------------------------------+
     SELECT 1 row in set (... sec)
 
-::
+The individual array elements can be selected from the ``landmarks`` column
+with ``landmarks[n]``, where ``n`` is the integer array index, like so::
 
-    cr> select race['name'], race['interests'] from locations
-    ... where 'books%' LIKE ANY(race['interests']);
-    +----------------+-----------------------------------------+
-    | race['name']   | race['interests']                       |
-    +----------------+-----------------------------------------+
-    | Bartledannians | ["netball", "books with 100.000 words"] |
-    +----------------+-----------------------------------------+
+    cr> select name, landmarks[1] from locations
+    ... where name = 'Frogstar';
+    +----------+--------------------------+
+    | name     | landmarks[1]             |
+    +----------+--------------------------+
+    | Frogstar | Total Perspective Vortex |
+    +----------+--------------------------+
     SELECT 1 row in set (... sec)
 
-It can also be used on arrays::
+.. NOTE::
 
-    cr> select name, race['interests'] from locations
-    ... where name = ANY(ARRAY['Bartledan', 'Algol'])
-    ... order by name asc;
-    +-----------+-----------------------------------------+
-    | name      | race['interests']                       |
-    +-----------+-----------------------------------------+
-    | Algol     | NULL                                    |
-    | Bartledan | ["netball", "books with 100.000 words"] |
-    +-----------+-----------------------------------------+
-    SELECT 2 rows in set (... sec)
+    The first index value is ``1``. The maximum array index is ``2147483648``.
+    Using an index greater than the array size results in a NULL value.
 
-This way it can be used as a shortcut for ``name = 'Bartledan' OR name =
-'Algol'`` or any other ANY comparison.
+Individual array elements can also be addressed in the :ref:`where clause
+<sql_dql_where_clause>`, like so::
 
-The ``ANY`` construct can be used in :ref:`sql_subquery_expressions` or
-:ref:`sql_array_comparisons`.
+    cr> select name, landmarks from locations
+    ... where landmarks[2] = 'Milliways';
+    +----------+-------------------------------------------+
+    | name     | landmarks                                 |
+    +----------+-------------------------------------------+
+    | Frogstar | ["Total Perspective Vortex", "Milliways"] |
+    +----------+-------------------------------------------+
+    SELECT 1 row in set (... sec)
 
+When using the ``=`` operator, as above, the value of the array element at
+index ``n`` is compared. To compare against *any* array element, see
+:ref:`sql_dql_any_array`.
 
-Negating ``ANY``
-----------------
+.. NOTE::
 
-One important thing to notice when using ANY is that negating the ANY operator
-does not behave as negating normal comparison operators.
+   You can nest arrays within arrays, but you can only have one subscript value
```

Comment:
No, you can **not** nest arrays with in arrays. Defining a columns as `array(array(text))` is not supported.